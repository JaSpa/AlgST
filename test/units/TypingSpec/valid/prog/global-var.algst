unit : ()
unit = ()

constUnit : forall a. a -> ()
constUnit [_] _ = unit
