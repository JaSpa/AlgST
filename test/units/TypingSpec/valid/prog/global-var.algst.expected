constUnit
|
+- type
|  |
|  `- Type.Forall 41#a:TU
|     |
|     `- Type.Arrow U
|        |
|        +- Type.Var 41#a
|        |  |
|        |  `- TU
|        |
|        `- Type.Unit
|
+- params
|  |
|  +- [39#_]
|  |
|  `- 40#_
|
`- definition
   |
   `- Exp.TypeAbs 39#_:TU
      |
      `- Exp.Abs U 40#_
         |
         +- Type.Var 39#_
         |  |
         |  `- TU
         |
         `- Exp.Var unit

unit
|
+- type
|  |
|  `- Type.Unit
|
+- params
|
`- definition
   |
   `- Exp.Lit
      |
      `- Lit.Unit

